src/CoreWCF.Http/tests/Helpers/AsyncOnlyStream.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace CoreWCF.Http.Tests.Helpers
+{
+    internal class AsyncOnlyStream : Stream
```

Comment:
This is an optional suggestion, so feel free to not do it. This could be simplified by copying DelegatingStream from the implementation into the tests. You can then make AsyncOnlyStream derive from it, and only override the methods you want different behavior for. You would only need to override the 3 methods.